crates/modelardb_server/tests/integration_test.rs

```diff
@@ -41,24 +41,96 @@ use sysinfo::{Pid, PidExt, System, SystemExt};
 
 use tokio::runtime::Runtime;
 use tonic::transport::Channel;
-use tonic::Request;
+use tonic::{Request, Response, Status, Streaming};
+
+const TABLE_NAME: &str = "table_name";
+const HOST: &str = "127.0.0.1";
+const PORT: u16 = 9999;
 
 /// The different types of tables used in the integration tests.
 enum TableType {
     NormalTable,
     ModelTable,
     ModelTableNoTag,
+    ModelTableAsField,
 }
 
-const TABLE_NAME: &str = "table_name";
-const HOST: &str = "127.0.0.1";
-const PORT: u16 = 9999;
+/// Instance of modelardbd used for testing. [`Child`] is wrapped in a struct to allow `drop()` to
+/// be implemented so modelardbd is stopped no matter if the a test succeeds, fails, or panics.
+struct ModelarDBD {
+    process: Child,
+}
+
+impl ModelarDBD {
+    /// Execute the binary with the server and return a handle to the process.
+    fn new(path: &Path) -> Self {
+        // Run modelardbd stdout and stderr is piped to /dev/null so the log messages and expected
+        // errors are not printed when all of the tests are run using the "cargo test" command.
+        let process = ModelarDBD::start_binary("modelardbd")
+            .arg(path)
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .spawn()
+            .expect("Failed to start Apache Arrow Flight Server");
+
+        // The thread needs to sleep to ensure that the server has properly started before sending
+        // streams to it.
+        thread::sleep(Duration::from_secs(5));
```

Comment:
I agree that would be the preferable approach and is also the approach I used when [implementing some of our utilities in Python](https://github.com/ModelarData/Utilities/blob/main/Evaluate-ModelarDB-Changes/main.py#L92). I think @agneborn98 looked into doing the same for the integration tests but was unable to properly kill `modelardbd` when its `stdout` or `stderr` was piped to the integration tests as it became defunct. I have added a low priority task to look into this again since it should be possible.

Reply:
Exactly, I did try this for a while. If it is any help, I believe windows was the OS that gave me the most trouble in killing the process(surprise....). This is because the program would panic if I tried to kill a PID which was already dead.